### PR TITLE
Fixes monkey mind magnification helmet

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Did you know that these don't work? It sucks. I bought one and was scammed! They're broken!

In my fit of rage over my broken product, I decided to fix it up! So that's what this PR is.

In my investigations, I found that:
A: The ghost polling was not set up correctly, meaning no ghosts were being polled, thus, the hat failed every time.
B: When a monke put the hat on themselves(they seem to prioritize these helmets), it would spam the buzz sound for it failing and then dust the hat several times. Thus breaking it even further given point A.

Not sure if these have ever worked correctly, but now they do! 

I removed the dusting part of it because that's actually dumb.

## Why It's Good For The Game

Having content work is good, right?

## Changelog
:cl:
fix: The monkey mind magnification helmet should now work correctly!
/:cl:

### CODE REVIEW

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -436,8 +435,11 @@
 	visible_message("<span class='warning'>[src] powers up!</span>")
 	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 	RegisterSignal(magnification, COMSIG_SPECIES_LOSS, .proc/make_fall_off)
+	INVOKE_ASYNC(src, /obj/item/clothing/head/helmet/monkey_sentience.proc/connect, user)
+
+/obj/item/clothing/head/helmet/monkey_sentience/proc/connect(mob/user)
 	polling = TRUE
-	var/list/candidates = pollCandidatesForMob("Do you want to play as a mind magnified monkey?", ROLE_SENTIENCE, null, ROLE_SENTIENCE, 50, magnification, POLL_IGNORE_SENTIENCE_POTION)
+	var/list/candidates = pollCandidatesForMob("Do you want to play as a mind magnified monkey?", ROLE_SENTIENCE, M = magnification, ignore_category = POLL_IGNORE_SENTIENCE_POTION) //SKYRAT EDIT CHANGE
```
Comment (JohnFulpWillard) [inline code/modules/clothing/head/helmet.dm:442]: Was this comment just a leftover?

Comment (Gandalf2k15) [reply code/modules/clothing/head/helmet.dm:442]: Whoops

Comment (Gandalf2k15) [reply code/modules/clothing/head/helmet.dm:442]: I PR'd it downstream, it's sitting open there too, thought I'd PR it upstream too.

**Comment by Time-Green** [comment]
I've messed around with them a bit, and apart from monkey players instantly attacking you/suiciding, they did work. They probably broke due to some refactor

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -456,14 +458,23 @@
 	if(policy)
 		to_chat(magnification, policy)
 	icon_state = "[icon_state]up"
+	REMOVE_TRAIT(magnification, TRAIT_PRIMITIVE, SPECIES_TRAIT) //Monkeys with sentience should be able to use less primitive tools.
 
 /obj/item/clothing/head/helmet/monkey_sentience/Destroy()
-	disconnect()
+	if(magnification)
```
Comment (AnturK) [inline code/modules/clothing/head/helmet.dm:464]: Shouldn't this be doing all the secondary disconnect effects if monkey still exists?

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -436,8 +435,11 @@
 	visible_message("<span class='warning'>[src] powers up!</span>")
 	playsound(src, 'sound/machines/ping.ogg', 30, TRUE)
 	RegisterSignal(magnification, COMSIG_SPECIES_LOSS, .proc/make_fall_off)
+	INVOKE_ASYNC(src, /obj/item/clothing/head/helmet/monkey_sentience.proc/connect, user)
```
Comment (AnturK) [inline code/modules/clothing/head/helmet.dm:438]: You don't need to type out full path if  it's the same as the type the call is in. `.proc/connect` should be enough here.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -456,14 +458,23 @@
 	if(policy)
 		to_chat(magnification, policy)
 	icon_state = "[icon_state]up"
+	REMOVE_TRAIT(magnification, TRAIT_PRIMITIVE, SPECIES_TRAIT) //Monkeys with sentience should be able to use less primitive tools.
 
 /obj/item/clothing/head/helmet/monkey_sentience/Destroy()
-	disconnect()
+	if(magnification)
+		ADD_TRAIT(magnification, TRAIT_PRIMITIVE, SPECIES_TRAIT)
```
Comment (Mothblocks) [inline code/modules/clothing/head/helmet.dm:465]: I don't like these side-effects on Destroy (this and the ghostize). Destroy should be nothing but destruction. This should be on `obj_destruction`.

File: code/modules/clothing/head/helmet.dm (hunk 2 above)
Comment (Mothblocks) [reply code/modules/clothing/head/helmet.dm:464]: :-1: Destroy should not be performing these kinds of side effects at all.

Comment (AnturK) [reply code/modules/clothing/head/helmet.dm:464]: I meant Unregisters and such, though you're right the sound/message stuff should be in destruction instead.

**Comment by 81Denton** [comment]
@Gandalf2k15 This PR has been stale for a while, please ping me or any other maintainer when you have time to address Mothblocks' review and the merge conflict.

**Pull request closed without merging**